packages/mongo-nav/src/org-nav/OrgNav.tsx

```diff
@@ -169,81 +179,154 @@ export default function OrgNav({
     OrgPaymentLabel.AdminSuspended,
   ];
 
+  let badgeItem: React.ReactElement | null = null;
+
+  if (
+    !isTablet &&
+    !isOnPrem &&
+    current?.paymentStatus &&
+    paymentVariant &&
+    (admin || paymentValues.includes(current.paymentStatus))
+  ) {
+    const badgeMargin = css`
+      margin-right: 25px;
+    `;
+
+    badgeItem = (
+      <Badge
+        className={badgeMargin}
+        variant={paymentVariant}
+        data-testid="org-nav-payment-status"
+      >
+        {current.paymentStatus.split('_').join()}
+      </Badge>
+    );
+  }
+
+  const renderedUserMenu = isOnPrem ? (
+    <div className={onPremMenuWrapper}>
```

Comment:
It'd be nice to have these as their own components but then we'd have to pass down all props again.

The context trick we talked about would be nice, but again, not a big deal at this point.

Reply:
I agree with you. Dave and I disagreed about this... @DesignerDave are you okay with me pulling this into a helper component now?

Reply:
As we chatted about on Slack, I don't feel strongly about this specifically so go for it. I'm just cognizant that the way we've broken this component up makes it somewhat difficult to follow right now.

Reply:
👍